src/main/java/com/redhat/devtools/lsp4ij/LanguageServersRegistry.java

```diff
@@ -449,6 +458,9 @@ private void removeAssociationsFor(LanguageServerDefinition definition) {
         request.serverDefinition().setInitializationOptionsContent(request.initializationOptionsContent());
         request.serverDefinition().setClientConfigurationContent(request.clientConfigurationContent());
 
+        // Increment the language server definition modification tracker
+        request.serverDefinition().incrementModificationCount();
```

Comment:
This increments the second modification tracker that's been added for the language server definition-level when it has changed.